Backend/TimeTracker.Api/Controllers/AuthController.cs

```diff
@@ -0,0 +1,202 @@
+        /// <returns>A response with an access key and a refresh key</returns>
+        [Route("/Login")]
+        [HttpPost]
+        public async Task<GenericResponseDTO<AccessKeysDTO>> Login(UserDTO loginData)
+        {
+            try
+            {
+                // Get user with a matching username and password hash
+                var hashedPassword = authHelper.GetPasswordHash(loginData.Password, configuration);
+                var curUser = await database.Users.FirstOrDefaultAsync(u => 
+                    u.Email == loginData.Email && u.Password.SequenceEqual(hashedPassword)
+                );
+
+                // If there was not a matching user then return an error
+                if(curUser == null)
+                {
+                    return new GenericResponseDTO<AccessKeysDTO>()
+                    {
+                        Success = false,
+                        Message = "Invalid username or password"
+                    };
+                }
+
+                // Generate the JWT and Refresh tokens and save the refresh token to the db
+                var accessToken = authHelper.GenerateJSONWebToken(curUser, configuration);
+                var refreshToken = authHelper.GenerateRefreshToken();
+
+                var userRefreshToken = new RefreshToken()
+                {
+                    Token = refreshToken,
+                    User = curUser
+                };
+
+                await database.AddAsync(userRefreshToken);
+                await database.SaveChangesAsync();
+
+                return new GenericResponseDTO<AccessKeysDTO>() 
+                {
+                    Success = true,
+                    Data = new AccessKeysDTO()
+                    {
+                         AccessToken = accessToken,
+                         RefreshToken = refreshToken
+                    }
+                };
+            }
+            catch
+            {
+                return new GenericResponseDTO<AccessKeysDTO>() 
+                {
+                    Success = false,
+                    Message = "An unknown error has occured"
+                };
+            }
+        }
+
+        /// <summary>
+        /// Endpoint to register a new user this defaults
+        /// the user to be a teacher
+        /// </summary>
+        /// <param name="registerData">This holds the registration information</param>
+        /// <returns>The Id of the user that was created</returns>
+        [Route("/Register")]
+        [HttpPost]
+        public async Task<GenericResponseDTO<int>> Register(UserDTO registerData)
+        {
+            try
+            {
+                // Make sure the password is valid
+                if (!authHelper.IsValidPassword(registerData.Password))
+                {
+                    return new GenericResponseDTO<int>() 
+                    {
+                        Success = false,
+                        Message = "Invalid password"
```

Comment:
Return information on why password was invalid

Reply:
Ya, I can add more details, really this error message should never be seen because it should be validated on the client-side before sending.

Reply:
Good point. We can keep validation on the frontend and keep it off the backend.

Reply:
This validation is really just preventing malicious users from doing weird things with the system.